core/lib/rom/schema.rb

```diff
@@ -136,14 +136,26 @@ def self.define(name, attributes: EMPTY_ARRAY, attr_class: Attribute, **options)
       ) { |schema| yield(schema) if block_given? }
     end
 
+    # Builds a representation of the information needed to create an
+    # attribute.
+    #
+    # This representation is consumed by `Schema.define` in order to create
+    # the actual attributes.
+    #
+    # @return [Hash] A hash with `:type` and `:options` keys.
+    #
+    # @api private
+    def self.build_attribute_info(type, options)
+      {
+        type: type,
+        options: options
+      }
+    end
+
     # @api private
     def self.attributes(attributes, attr_class)
       attributes.map do |attr|
-        if attr.is_a?(Hash)
-          attr_class.new(attr[:type], attr.fetch(:options, EMPTY_HASH))
-        else
-          attr_class.new(attr)
-        end
+        attr_class.new(attr[:type], attr.fetch(:options))
```

Comment:
I'll try to explain here the issue with backward compatibility. The DSL is calling `Schema.define`, which in turn is calling `Schema.attributes`. Before these changes, latter expected the attributes to be an array of types. Now it expects it to be an array of hashes `:type`/`:options`. Not sure if it is an issue with schema plugins or something.

Reply:
`Schema.attributes` is a private API, no need to worry about it